pkg/services/teams-workflows.go

```diff
@@ -0,0 +1,519 @@
+package services
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/url"
+	"regexp"
+	"strings"
+	texttemplate "text/template"
+
+	httputil "github.com/argoproj/notifications-engine/pkg/util/http"
+)
+
+var (
+	// WorkflowsURLPattern matches Microsoft Teams Workflows webhook URLs (Power Automate)
+	// Matches: api.powerautomate.com, api.powerplatform.com, flow.microsoft.com, or webhook.office.com/workflows
+	// Also matches URLs with /powerautomate/ in the path (Power Platform URLs)
+	WorkflowsURLPattern = regexp.MustCompile(`(?i)(api\.(powerautomate|powerplatform)\.com|flow\.microsoft\.com|webhook\.office\.com/workflows|/powerautomate/)`)
+)
+
+type TeamsWorkflowsNotification struct {
+	Template        string `json:"template,omitempty"`
+	Title           string `json:"title,omitempty"`
+	Summary         string `json:"summary,omitempty"`
+	Text            string `json:"text,omitempty"`
+	ThemeColor      string `json:"themeColor,omitempty"`
+	Facts           string `json:"facts,omitempty"`
+	Sections        string `json:"sections,omitempty"`
+	PotentialAction string `json:"potentialAction,omitempty"`
+	// AdaptiveCard specifies the Adaptive Card JSON payload (alternative to messageCard)
+	AdaptiveCard string `json:"adaptiveCard,omitempty"`
+	// CardFormat specifies which card format to use: "messageCard" (default) or "adaptiveCard"
+	CardFormat string `json:"cardFormat,omitempty"`
+}
+
+func (n *TeamsWorkflowsNotification) GetTemplater(name string, f texttemplate.FuncMap) (Templater, error) {
+	template, err := texttemplate.New(name).Funcs(f).Parse(n.Template)
```

Comment:
I think we can save some CPU cycles and memory if we init the template once like this
`tmpl := texttemplate.New(name).Funcs(f)`

and then re-use the variable to the parsing of each field. 

```
	template, err := tmpl.Parse(n.Template)
	if err != nil {
		return nil, fmt.Errorf("error in '%s' teams-workflows.template : %w", name, err)
	}
```

etc

WDYT: